### DISCUSSION TITLE
Spring Boot application without java-agent

### DISCUSSION THREAD

**Discussion by xabierMoreno** [open-telemetry/opentelemetry-java-instrumentation#2786] [Q&A]
Hi,
First of all, I would like to say that the project is really good.  
I am working in a spring boot application with microservices . I´ve tested the auto-instrumentation and It works very well. However,  I cannot use the java-agent "opentelemetry-javaagent-all.jar" in my application, so  if I understand correctly the other possibility will be doing the manual-instrumentation but that will require to create the span, send to services and  finally add in the headers for the next microservice request and add this logic to all of our methods.

My question is the following,  is there any way that  I can import some dependencies in my application and have the same functionality as autoinstrumentation java-agent without writing any code? I've tried adding 'opentelemetry-spring-boot-autoconfigure' but it will only create one span per request and it won´t have any information about the nested spans.

Looking forward to hearing from you,
Xabi

**Comment by iNikem** [comment]
I believe Spring Boot autoconfiguration module does not work for you because of https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/2771
